### PR TITLE
Add Authentication

### LINKED ISSUE #95: Authentication: API authorization

### LINKED ISSUE #96: Authentication: Sign in page

### PR DESCRIPTION
**Pull request by MrF3lix**
## Related Issues
 
- Closes #95
- Closes #96

## Definition of Done

- [ ] User Story
  - [ ] All Acceptance Criteria met
- [ ] Code
  - [ ] No more code needed
  - [ ] No known bugs
- [ ] Clean Code
  - [ ] Comments have been created for new code
  - [ ] ESLinter found no issues (editor & executor only)
  - [ ] No unavoidable Code Smells
- [ ] Testing
  - [ ] Existing Tests pass
  - [ ] New Unit Tests created
  - [ ] New Unit Tests pass
- [ ] SonarCloud Quality Gate passed

## Definition of more than Done (optional)

- [ ] Testing
  - [ ] New Integration Tests created
  - [ ] New Integration Tests passed

### CODE REVIEW

File: src/api/Flooq.Api/Controllers/UserController.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+namespace Flooq.Api.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class UserController : ControllerBase
+{
+  private readonly IUserService _userService;
+
+  public UserController(IUserService userService)
```
Comment (DuplosFidibuss) [inline src/api/Flooq.Api/Controllers/UserController.cs:13]: Do we still need this controller?

File: src/api/Flooq.Api/Program.cs [hunk 2]
```diff
@@ -1,23 +1,97 @@
...
 using Microsoft.EntityFrameworkCore;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.OpenApi.Models;
 
 var builder = WebApplication.CreateBuilder(args);
 
 builder.Services.AddDbContext<FlooqContext>(options => options.UseNpgsql(builder.Configuration.GetConnectionString("FlooqDatabase")));
+// builder.Services.AddIdentity<IdentityUser, IdentityRole>().AddEntityFrameworkStores<FlooqContext>();
 builder.Services.AddControllers();
 builder.Services.AddEndpointsApiExplorer();
```
Comment (DuplosFidibuss) [inline src/api/Flooq.Api/Program.cs:14]: Dead code?

File: src/api/Flooq.Api/Services/IUserService.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+
+namespace Flooq.Api.Services;
+
+public interface IUserService
+{
+  Task<ActionResult<IdentityUser>> GetUserById(string? id);
+  
+  Task<ActionResult<IdentityUser>> GetUserByEmail(string? email);
+
```
Comment (DuplosFidibuss) [inline src/api/Flooq.Api/Services/IUserService.cs:12]: Like the controller: is it still needed? If yes, comments should be added.

**Review by DuplosFidibuss — APPROVED** [review]

File: src/api/Flooq.Api/Services/IUserService.cs (hunk 3 above)
Comment (MrF3lix) [reply src/api/Flooq.Api/Services/IUserService.cs:12]: Yep, still needed

**Pull request merged**